packages/web/src/pages/sign-up-page/components/HandleField.tsx

```diff
@@ -0,0 +1,115 @@
+
+import { useDebouncedCallback } from '@audius/common'
+import { TextLink } from '@audius/harmony'
+import { useField, useFormikContext } from 'formik'
+
+import {
+  HarmonyTextField,
+  HarmonyTextFieldProps
+} from 'components/form-fields/HarmonyTextField'
+import { ToastContext } from 'components/toast/ToastContext'
+
+import { errorMessages } from '../utils/handleSchema'
+import { socialMediaMessages } from '../utils/socialMediaMessages'
+
+import { SignupFlowInstagramAuth } from './SignupFlowInstagramAuth'
+import { SignupFlowTikTokAuth } from './SignupFlowTikTokAuth'
+import { SignupFlowTwitterAuth } from './SignupFlowTwitterAuth'
+
+const messages = {
+  handle: 'Handle',
+  linkToClaim: 'Link to claim.'
+}
+
+const handleAuthMap = {
+  [errorMessages.twitterReservedError]: SignupFlowTwitterAuth,
+  [errorMessages.instagramReservedError]: SignupFlowInstagramAuth,
+  [errorMessages.tiktokReservedError]: SignupFlowTikTokAuth
+}
+
+type HandleValues = {
+  handle: string
+}
+
+type HandleFieldProps = Partial<HarmonyTextFieldProps> & {
+  onCompleteSocialMediaLogin?: (info: {
+    requiresReview: boolean
+    handle: string
+    platform: 'twitter' | 'instagram' | 'tiktok'
+  }) => void
+}
+
+export const HandleField = (props: HandleFieldProps) => {
+  const { onCompleteSocialMediaLogin, ...other } = props
+  const [{ value: handle }, { error }, { setError }] = useField('handle')
+  const { validateForm } = useFormikContext<HandleValues>()
+
+  const { toast } = useContext(ToastContext)
+
+  const debouncedValidate = useDebouncedCallback(
+    validateForm,
+    [validateForm],
+    1000
+  )
+
+  useEffect(() => {
+    debouncedValidate({ handle })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [debouncedValidate, handle])
+
+  const handleVerifyHandleError = useCallback(() => {
+    setError(socialMediaMessages.verificationError)
+  }, [setError])
+
+  const handleLoginSuccess = useCallback(
+    ({
+      handle,
+      requiresReview,
+      platform
+    }: {
+      requiresReview: boolean
+      handle: string
+      platform: 'twitter' | 'instagram' | 'tiktok'
+    }) => {
+      toast(socialMediaMessages.socialMediaLoginSucess(platform))
+      onCompleteSocialMediaLogin?.({
+        handle,
+        requiresReview,
+        platform
+      })
+    },
+    [onCompleteSocialMediaLogin, toast]
+  )
+
+  const AuthComponent = error ? handleAuthMap[error] : undefined
+
+  const helperText = error ? (
+    <>
+      {error}{' '}
+      {onCompleteSocialMediaLogin && AuthComponent ? (
+        <TextLink variant='visible' asChild>
+          <AuthComponent
+            onFailure={handleVerifyHandleError}
+            onSuccess={handleLoginSuccess}
+          >
+            <span>{messages.linkToClaim}</span>
+          </AuthComponent>
+        </TextLink>
+      ) : null}
+    </>
+  ) : null
+
+  return (
+    <HarmonyTextField
+      name='handle'
+      label={messages.handle}
+      error={error && handle}
```

Comment:
Is `handle` the correct value for the `error` prop?

Reply:
so thats just boolean, where, errormessage is the string, so i think we good

Reply:
Technically it is passing the handle string to `error`, which then would get cast to a boolean probably. But I'm not sure I quite follow, where is errorMessage being set? and it doesn't look like `HarmonyTextField` uses the `error` prop at all, because it overrides it with `hasError`. I may just not have enough context

Reply:
ah sorry, so "helperText" is what appears at the bottom, and "error" determines if it's red. you can pass elements to text-field that override default implementation, which it looks like we are trying to do for error and helperText, since this field is a bit weird in that it runs the full form validation. I'd like to see if we can improve this field to just run validation on it specifically, rather than the entire form.... but as it stands, that's why things "work" as expected

Reply:
Ah gotcha, ok all makes sense. thanks!